srml/executive/src/lib.rs

```diff
@@ -339,7 +344,11 @@ impl<
 				provides,
 				longevity: TransactionLongevity::max_value(),
 			}
+		} else if let Some(valid_transaction) = Validatable::validate_transaction(xt.call()) {
+			// TODO TODO: do we want this order ? (first if signed else try validate)
+			valid_transaction
 		} else {
+			// TODO TODO: Change those errors.
```

Comment:
no its correct: if a module wants to return a specific invalidity it returns Some(Invalid(..))